ethcore/src/service.rs

```diff
@@ -130,12 +132,16 @@ impl IoHandler<NetSyncMessage> for ClientIoHandler {
 	}
 
 	#[cfg_attr(feature="dev", allow(single_match))]
-	fn message(&self, io: &IoContext<NetSyncMessage>, net_message: &NetSyncMessage) {
-		if let UserMessage(ref message) = *net_message {
+	fn message(&self, io: &IoContext<NetSyncMessage>, net_message: &mut NetSyncMessage) {
```

Comment:
It is guaranteed in rust, the handler can move out message content, but never leave it in incorrect state. Moving the whole message is not possible since there are multiple handlers, each interested in a particular set of message types.